jmxtrans-output/jmxtrans-output-influxdb/src/main/java/com/googlecode/jmxtrans/model/output/InfluxDbWriter.java

```diff
@@ -161,44 +166,65 @@ public InfluxDbWriter(
 	 */
 	@Override
 	public void doWrite(Server server, Query query, Iterable<Result> results) throws Exception {
+
 		// Creates only if it doesn't already exist
 		if (createDatabase) influxDB.createDatabase(database);
 		BatchPoints.Builder batchPointsBuilder = BatchPoints.database(database).retentionPolicy(retentionPolicy)
 				.tag(TAG_HOSTNAME, server.getSource());
 
+		//Custom tags
 		for(Map.Entry<String,String> tag : tags.entrySet()) {
 			batchPointsBuilder.tag(tag.getKey(),tag.getValue());
 		}
-
-		BatchPoints batchPoints = batchPointsBuilder.consistency(writeConsistency).build();
 		
 		ImmutableList<String> typeNamesParam = null;
 		// if not typeNamesAsTag, we concat typeName in values.
 		if (!typeNamesAsTags) {
 			typeNamesParam = this.typeNames;
 		}
 		
+		//Creates a tag map, associated by the typeName.
+		//Results fields with the same typeName will receive the same tag values.
+		HashMap<String,HashMap<String, String>> attrTagByTypeName = newHashMap();
+		if (!attributesAsTags.isEmpty()) {
```

Comment:
Extracting a method may make it more readable.
Is it possible to convert `HashMap<HashMap>` into a `Map<Map>`?

Reply:
No problem, it will be done in the next commit.